### ISSUE TITLE
The voice permissions access can't be denied and the prompt is redisplayed automatically

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#3491]
**[Notes]:**
- This issue is more damaging if the "Don't ask again" checkbox is checked before clicking the "Deny" button, resulting in the app or voice search being broken, even after restarting the device. For example on Pico Neo2, UI glitches are displayed, making the browser unusable or on Oculus Quest, a black screen is displayed each time the Microphone button is clicked.

**[Affected Versions]:**
- Firefox Reality version 11-141621406 (c7dd36c).

**[Affected Platforms]:** 
- Oculus Quest - Android 7.1.1
- HTC Vive Focus - Android 7.1.2
- Pico Neo2 - Android 8.1.0

**[Prerequisites]:**
- The Firefox Reality browser is open.

**[Steps to reproduce]:**
1. Click the microphone button from the Awesomebar.
2. Dismiss the voice collection prompt.
3. Click the "Deny" button from the voice permissions prompt.
4. Observe the behavior.

**[Expected results]:**
- The prompt is dismiss and permissions are denied.

**[Actual results]:**
- The prompt remains displayed.

**[Additional Notes]:**
- This issue is not reproducible if the "Allow" button from the permissions prompt is clicked.
- This issue is not reproducible when denying Storage or Location permissions.
- The HTC Vive Focus device, seems to be affected by this issue only after resuming the app, where the permissions prompt and voice search prompt are redisplayed automatically.
- Attached are screen recordings of the issue:
	- Screen Recording Quest: [link](https://drive.google.com/file/d/1547tYfS33wEbNqfapCdHh0G2_nASvNq_/view?usp=sharing).
	- Screen Recording Pico Neo2: [link](https://drive.google.com/file/d/1xYVkROZ9WiIV2NLChEfw-JXymJXyzBGN/view?usp=sharing).
	- Screen Recording HTC Vive Focus: [link](https://drive.google.com/file/d/1QUK6xnby8pfj3sLSmH97EkFQXpVJXXrR/view?usp=sharing).

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and it's no longer reproducible using Firefox Reality 11-141672019 (418d959). The permissions are now denied and the prompt is not redisplayed.
However, a new issue can be observed on the HTC devices where the permission prompt is no longer displayed, if the app is quited and resumed while the prompt is displayed. This is tracked in: #3511.

Tested on HTC Vive Focus (Android 7.1.2), Oculus Quest (Android 7.1.1), and Pico Neo 2 (Android - 8.1.0) devices.
